### PR TITLE
Stop installing kubeadm types in the generic, legacy scheme

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

Right now the kubeadm API types are (wrongly from a correctness perspective) installed in the "catch-all" and now legacy `pkg/api/legacyscheme`. Instead, we should use our own, kubeadm-specific `scheme` where our API types are registered. Doing this also cuts one dependency on the internal core API types.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes/community/pull/2131

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```
@kubernetes/sig-cluster-lifecycle-pr-reviews @sttts @liztio

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/fuzzer/fuzzer_test.go [hunk 1]
```diff
@@ -14,23 +14,26 @@ See the License for the specific language governing permissions and
...
 )
 
-func init() {
-	Install(legacyscheme.Scheme)
-}
-
-// Install registers the API group and adds types to a scheme
-func Install(scheme *runtime.Scheme) {
+// installAPIsToScheme registers the API group and adds types to a scheme
+func installAPIsToScheme(scheme *runtime.Scheme) {
```
Comment (sttts) [inline cmd/kubeadm/app/apis/kubeadm/fuzzer/fuzzer_test.go]: why a local installer func? Why not the one shared by the kubeadm scheme?

**Review by liztio — COMMENTED** [review]
Seems like I've still got a lot to learn about apimachinery. Why is /install no longer needed?

File: cmd/kubeadm/app/apis/kubeadm/fuzzer/fuzzer_test.go (hunk 1 above)
Comment (luxas) [reply cmd/kubeadm/app/apis/kubeadm/fuzzer/fuzzer_test.go]: Good feedback! I'm now using `scheme.AddToScheme` instead, with the extra registrations added

**Comment by liztio** [comment]
Ah, I get it now. /install installs the kinds into the global scheme, so we don't need it anymore.

**Comment by luxas** [comment]
So earlier we used the `legacyscheme` for all API machinery operations involving kubeadm API types. Hence we needed to install our API types in the `legacyscheme`. That was what the `install` package was used for, it registered the kubeadm API types in the catch-all scheme. The `legacyscheme` was also used for normal API machinery operations, like unmarshalling ConfigMaps, etc. from bytes.

Now, we register the kubeadm API types in the kubeadm scheme only, and _only_ use the kubeadm scheme everywhere when dealing with kubeadm API machinery operations. When dealing with "normal" types, we use the `client-go` scheme. Makes sense?

File: cmd/kubeadm/app/cmd/phases/addons_test.go [hunk 2]
```diff
@@ -20,7 +20,6 @@ import (
 	"testing"
 
 	// required for triggering api machinery startup when running unit tests
-	_ "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/install"
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/cmd/phases/addons_test.go]: Delete comment as well? (Same applies to other test file in the same folder

Comment (luxas) [reply cmd/kubeadm/app/cmd/phases/addons_test.go]: Done

**Comment by fabriziopandini** [comment]
/lgtm 
I let @timothysc to say a final word on this one

File: cmd/kubeadm/app/phases/addons/dns/dns.go [hunk 3]
```diff
@@ -31,12 +31,12 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	kuberuntime "k8s.io/apimachinery/pkg/runtime"
 	clientset "k8s.io/client-go/kubernetes"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
```
Comment (timothysc) [inline cmd/kubeadm/app/phases/addons/dns/dns.go]: I wouldn't call this clientset, b/c it's int the interface layer iirc.  clientscheme is sufficient.

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 

Minor nits but not worth holding this up.

**Pull request merged**